### ISSUE TITLE
[BUG] avoid-unrelated-type-assertions has false positives

### ISSUE THREAD

**Issue by janknips** [dart-code-checker/dart-code-metrics#940]
* **Dart code metrics version: ^4.16.0**
* **Dart sdk version:  2.17.6**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel unknown, 3.0.5, on macOS 12.4 21F79 darwin-x64, locale en-DE)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] Android Studio (version 2021.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2022.1.4)
[✓] IntelliJ IDEA Ultimate Edition (version 2022.1.4)
[✓] IntelliJ IDEA Ultimate Edition (version 2022.1.3)
[✓] Connected device (2 available)
[✓] HTTP Host Availability

• No issues found!
```

</details>

**What did you do? Please include the source code example causing the issue.**

```dart
Fruit _fruit = FruitAndVeggie();

if(_fruit is Veggie) {
print('can be true but analyzer says otherwise');
}

class FruitAndVeggie implements Fruit, Veggie {}

```

**What did you expect to happen?** No linter warning or a warning saying: "Note: there might be false positives if another type is implemented"

**What actually happened?** Linter shows a warning

**Are you willing to submit a pull request to fix this bug?** No/maybe

**Comment by incendial** [comment]
This is an expected false-positive, unfortunately, sometimes it's impossible to calculate the relation between two types. Closing as won't fix.
